internal/libdrop_event_to_string_test.go

```diff
@@ -0,0 +1,101 @@
+package internal
+
+import (
+	"fmt"
+	"testing"
+
+	norddrop "github.com/NordSecurity/libdrop-go/v7"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEventToStringWithWorkingMarshaller(t *testing.T) {
+	event := norddrop.Event{
+		Kind: norddrop.EventKindRequestQueued{
+			Peer:       "12.12.12.11",
+			TransferId: "c13c619c-c70b-49b8-9396-72de88155c43",
+			Files: []norddrop.QueuedFile{
+				{
+					Id:   "file1ID",
+					Path: "testfile-small",
+					Size: 100,
+				},
+				{
+					Id:   "file2ID",
+					Path: "testfile-big",
+					Size: 1000,
+				},
+				{
+					Id:   "file3ID",
+					Path: "file3.txt",
+					Size: 1000,
+				},
+			},
+		},
+	}
+
+	expected := `{
+    "Timestamp": 0,
+    "Kind": {
+      "Peer": "12.12.12.11",
+      "TransferId": "c13c619c-c70b-49b8-9396-72de88155c43",
+      "Files": [
+        {
+          "Id": "file1ID",
+          "Path": "testfile-small",
+          "Size": 100,
+          "BaseDir": null
+        },
+        {
+          "Id": "file2ID",
+          "Path": "testfile-big",
+          "Size": 1000,
+          "BaseDir": null
+        },
+        {
+          "Id": "file3ID",
+          "Path": "file3.txt",
+          "Size": 1000,
+          "BaseDir": null
+        }
+      ]
+    }
+  }`
+
+	assert.JSONEq(t, expected, eventToString(event, jsonMarshaler{}))
+}
+
+func TestEventToStringWithBrokenMarshaler(t *testing.T) {
+	event := norddrop.Event{
+		Kind: norddrop.EventKindRequestQueued{
+			Peer:       "12.12.12.12",
+			TransferId: "c13c619c-c70b-49b8-9396-72de88155c43",
+			Files: []norddrop.QueuedFile{
+				{
+					Id:   "file1ID",
+					Path: "testfile-small",
+					Size: 100,
+				},
+				{
+					Id:   "file2ID",
+					Path: "testfile-big",
+					Size: 1000,
+				},
+				{
+					Id:   "file3ID",
+					Path: "file3.txt",
+					Size: 1000,
+				},
+			},
+		},
+	}
+
+	expected := "norddrop.EventKindRequestQueued"
+
+	assert.Equal(t, expected, eventToString(event, brokenMarshaler{}))
+}
```

Comment:
This could be a single parametrized table driven test with 2 cases.

Reply:
I was thinking about that as well, but the assertion is different, because in one case I'm comparing plain string, but in other I'm comparing JSON strings. I could to some check in the assertion part and when it's not a JSON then compare plain strings, but it's bit ugly and introduces not needed complexity IMO